packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -191,7 +313,7 @@ export default class ScriptTransformer {
       return cached;
     }
 
-    let transformer: Transformer = require(transformPath);
+    let transformer: SyncTransformer = require(transformPath);
```

Comment:
we might need to refactor to load all transformers in some warmup step later so people can write their transformers in ESM (which cannot be `require`d). Not sure if we should stick the refactor in here as well... Thoughts?

(To be clear, the transformer _itself_ can be written in ESM but still expose sync transformations. Concrete example: #11081. While that talks about doing the `import()` in an async thing before using and the transformer itself being CJS, I think people will expect to be able to write ESM directly)

Reply:
I've implemented this in #11163